docgen/parser/parser.go

```diff
@@ -155,14 +155,63 @@ func parseValues(path string, comments map[string]*Comment, values []*google_pro
 	return res
 }
 
+// extractMapFields generates an array of fields that correspond to map type fields for messages, which are stored in
+// the NestedType field of the DescriptorProto struct
+func extractMapFields(message *google_protobuf.DescriptorProto, path string, comments map[string]*Comment) []*Field {
+	fields := make([]*Field, len(message.NestedType))
+	for i, n := range message.GetNestedType() {
+		// a nested type of map will always have two fields for key and value
+		if len(n.GetField()) != 2 {
+			continue
```

Comment:
this should be a panic, because of the comment above. you can also just make the comment be the panic message, and simplify the code.

if you do a `continue`, we might be silently ignoring data.